aws-lc-rs/src/agreement/ephemeral.rs

```diff
@@ -0,0 +1,462 @@
+                        };
+                        let my_public = test_case.consume_bytes("MyQ");
+                        let output = test_case.consume_bytes("Output");
+
+                        assert_eq!(my_private.algorithm(), alg);
+
+                        let computed_public = my_private.compute_public_key().unwrap();
+                        assert_eq!(computed_public.as_ref(), &my_public[..]);
+
+                        assert_eq!(my_private.algorithm(), alg);
+
+                        let result = agreement::agree_ephemeral(
+                            my_private,
+                            &peer_public,
+                            (),
+                            |key_material| {
+                                assert_eq!(key_material, &output[..]);
+                                Ok(())
+                            },
+                        );
+                        assert_eq!(
+                            result,
+                            Ok(()),
+                            "Failed on private key: {:?}",
+                            test::to_hex(my_private_bytes)
+                        );
+                    }
+
+                    Some(_) => {
+                        fn kdf_not_called(_: &[u8]) -> Result<(), ()> {
+                            panic!(
+                                "The KDF was called during ECDH when the peer's \
+                         public key is invalid."
+                            );
+                        }
+                        let dummy_private_key =
+                            agreement::EphemeralPrivateKey::generate(alg, &rng)?;
+                        assert!(agreement::agree_ephemeral(
+                            dummy_private_key,
+                            &peer_public,
+                            (),
+                            kdf_not_called
+                        )
+                        .is_err());
+                    }
+                }
+
+                Ok(())
+            },
+        );
+    }
+
+    fn h(s: &str) -> Vec<u8> {
+        match test::from_hex(s) {
+            Ok(v) => v,
+            Err(msg) => {
+                panic!("{msg} in {s}");
+            }
+        }
+    }
+
+    fn alg_from_curve_name(curve_name: &str) -> &'static agreement::Algorithm {
+        if curve_name == "P-256" {
+            &agreement::ECDH_P256
+        } else if curve_name == "P-384" {
+            &agreement::ECDH_P384
+        } else if curve_name == "P-521" {
+            &agreement::ECDH_P521
+        } else if curve_name == "X25519" {
+            &agreement::X25519
+        } else {
+            panic!("Unsupported curve: {curve_name}");
+        }
+    }
+
+    fn x25519(private_key: &[u8], public_key: &[u8]) -> Vec<u8> {
+        x25519_(private_key, public_key).unwrap()
+    }
+
+    fn x25519_(private_key: &[u8], public_key: &[u8]) -> Result<Vec<u8>, Unspecified> {
```

Comment:
Personally not a fan of the `_` suffix, can we call it `try_x25519`?

```suggestion
    fn try_x25519(private_key: &[u8], public_key: &[u8]) -> Result<Vec<u8>, Unspecified> {
```

Or just combine the functions as it appears `x25519_` is not used by the tests directly?